packages/p2p-v2/src/hooks/useQueryString.ts

```diff
@@ -0,0 +1,109 @@
+import { useEffect, useRef, useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import { useEventListener } from 'usehooks-ts';
+
+declare global {
+    interface WindowEventMap {
+        queryChange: CustomEvent;
+    }
+}
+
+type TSetQueryStringOptions = {
+    clearAllQueries?: boolean;
+};
+
+type TQueryString = {
+    advertId?: string;
+    form?: string;
+    modal?: string;
+    paymentMethodId?: string;
+    tab?: string;
+};
+
+function mergeParams(params1: URLSearchParams, params2: URLSearchParams) {
+    const payload1 = paramsToPayload(params1);
+    const payload2 = paramsToPayload(params2);
+    const formattedParams = payloadToParams({
+        ...payload1,
+        ...payload2,
+    });
+
+    return formattedParams;
+}
+
+const payloadToParams = (payload: TQueryString) => new URLSearchParams(Object.entries(payload));
+const paramsToPayload = (params: URLSearchParams) => Object.fromEntries(params.entries());
+const queryEvent = new Event('queryChange');
+
+// TODO: Add validations for query strings once My Profile page is merged
+// type TQueryStringConfig = {
+//     tab?: {
+//         default: string;
+//         values: string[];
+//     };
+// };
+
+function useQueryString() {
```

Comment:
what is this hook for?

Reply:
I'll add some TSdocs, its for syncing the url params like `p2p/v2/my-profile?tab=Stats` to rendering the correct tabs/form/screens

Reply:
really good stuff man 🔥 yes please add some tsdocs for these